### ISSUE TITLE
Not all SpecifiedTradeSettlementPaymentMeans type codes are mapped

### ISSUE THREAD

**Issue by shenazz** [phax/en16931-cii2ubl#34]
Not all SpecifiedTradeSettlementPaymentMeans type codes from CII are mapped in UBL as PaymentMeans. 

For eg, if CII file has the below type code for SpecifiedTradeSettlementPaymentMeans, PaymentMeans is missing in UBL.

![image](https://github.com/user-attachments/assets/beb7f9aa-e0f1-4eac-90c0-7d9531c56d18)

I believe following are the supported SpecifiedTradeSettlementPaymentMeans type codes in CII - https://portal3.gefeg.com/projectdata/invoice/deliverables/installed/publishingproject/zugferd%202.0.1%20-%20facturx%201.03/en%2016931%20%E2%80%93%20facturx%201.03%20%E2%80%93%20zugferd%202.0.1%20-%20basic.scm/html/de/021.htm?https://portal3.gefeg.com/projectdata/invoice/deliverables/installed/publishingproject/zugferd%202.0.1%20-%20facturx%201.03/en%2016931%20%E2%80%93%20facturx%201.03%20%E2%80%93%20zugferd%202.0.1%20-%20basic.scm/html/de/02134.htm

**Comment by phax** [comment]
So you are referring to  `BT-81` "Payment means type code".
In UBL it is mapped to `/Invoice/cac:PaymentMeans/cbc:PaymentMeansCode`
In CII it is mapped to `/rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeSettlement/ram:SpecifiedTradeSettlementPaymentMeans/ram:TypeCode`

`BT-81` description: 
> Entries from the UNTDID 4461 code list shall be used. Distinction should be made between SEPA and non-SEPA payments, and between credit payments, direct debits, card payments and other instruments.

The D16B version of the code list can be found at https://unece.org/fileadmin/DAM/trade/untdid/d16b/tred/tred4461.htm

Both CII and UBL mappings of EN 163931 point to code list 4461, so no additional mapping is required

**Comment by shenazz** [comment]
Thank you for the quick response.

Sorry if the problem statement in my earlier post was unclear. I will try to explain again.

When I convert an invoice in CII format to UBL and if the CII file has `ram:SpecifiedTradeSettlementPaymentMeans/ram:TypeCode` value as 10 (In Cash), `/Invoice/cac:PaymentMeans` is missing in the converted UBL file.

In the source code (CIIToUBL21Converter.java) I can see that `/Invoice/cac:PaymentMeans` is added only if the following conditions are met

```
if (bIsBG17 || bIsBG18 || bIsBG19 || isPaymentMeansCodeOtherKnown (sTypeCode))
      aPaymentMeansHandler.accept (aUBLPaymentMeans);
else
      aErrorList.add (buildError (null,
                                  "Failed to determine a supported Payment Means Type from code '" + sTypeCode + "'"));
```

`isPaymentMeansCodeOtherKnown ` does not include 10.

**Comment by phax** [comment]
Thanks @shenazz for bringing that up. It is fixed in version 2.2.6 of this project.
